code/search/SearchCriteria.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+class SearchCriteria implements SearchCriteriaInterface
+{
+    /**
+     * "AND" is a protected PHP7 keyword.
+     *
+     * @param string
+     */
+    const UND = 'AND';
+
+    /**
+     * "OR" is a protected PHP7 keyword.
+     *
+     * @param string
+     */
+    const ODER = 'OR';
```

Comment:
I dislike the naming of these constants. Can you use `CONJUNCTION_OR` and `CONJUNCTION_AND` instead maybe?